bpf/bpf_host.c

```diff
@@ -360,18 +359,11 @@ handle_ipv6(struct __ctx_buff *ctx, __u32 secctx, const bool from_host)
 	}
 #endif /* ENABLE_NODEPORT */
 
-	nexthdr = ip6->nexthdr;
-	hdrlen = ipv6_hdrlen(ctx, l3_off, &nexthdr);
-	if (hdrlen < 0)
-		return hdrlen;
-
-#ifdef HANDLE_NS
-	if (!skip_redirect && unlikely(nexthdr == IPPROTO_ICMPV6)) {
-		ret = icmp6_handle(ctx, ETH_HLEN, ip6, METRIC_INGRESS);
-		if (IS_ERR(ret))
+	if (!skip_redirect) {
+		ret = icmp6_host_handle(ctx, ip6->nexthdr);
```

Comment:
It's a bit counter-intuitive to call into ICMPv6 handling before checking whether the packet is ICMPv6, but looking at the code the implementation remains the same for non-ICMPv6 traffic so LGTM.